snowplow-tracker/src/androidTest/java/com/snowplowanalytics/snowplow/tracker/SubjectTest.java

```diff
@@ -114,5 +130,27 @@ public void testSubjectUserIdCanBeUpdated() {
         assertNull(tracker.getSubject().getUserId());
     }
 
+    @Test
+    public void testAnonymisesUserIdentifiers() {
+        Subject subject = getSubject();
+        subject.setUserId("uid-test");
+        subject.setDomainUserId("duid-test");
+        subject.setNetworkUserId("nuid-test");
+        subject.setIpAddress("127.0.0.1");
+        assertNull(subject.getSubject(true).get("uid"));
+        assertNull(subject.getSubject(true).get("duid"));
+        assertNull(subject.getSubject(true).get("tnuid"));
+        assertNull(subject.getSubject(true).get("ip"));
+    }
+
+    // Helper Methods
+
+    private Subject getSubject() {
```

Comment:
Nit: This could be a good opportunity to change the name of this helper so it's not the same as Subject `getSubject()`, I was briefly confused

Reply:
Good point, fixed!